From cccf831ed7c83410c7f6cec2a43e71e9c4278b4c Mon Sep 17 00:00:00 2001
From: "Thomas E. Dickey" <dickey@invisible-island.net>
Date: Sat, 19 May 2018 22:19:12 +0000
Subject: [PATCH] ncurses 6.1 - patch 20180519

+ formatting fixes for manpages, regenerate man-html documentation.
+ trim spurious whitespace from tmux in 2018-02-24 changes;
  fix some inconsistencies in/between tmux- and iterm2-entries for SGR
  (report by C Anthony Risinger)
+ improve iterm2 using some xterm features which it has adapted -TD
+ add check in pair_content() to handle the case where caller asks
  for an uninitialized pair (Debian #898658).
---
 NEWS                                  | 11 +++++-
 VERSION                               |  2 +-
 dist.mk                               |  4 +-
 doc/html/man/adacurses6-config.1.html |  2 +-
 doc/html/man/captoinfo.1m.html        |  2 +-
 doc/html/man/clear.1.html             |  6 +--
 doc/html/man/curs_addch.3x.html       |  4 +-
 doc/html/man/form.3x.html             |  2 +-
 doc/html/man/form_driver.3x.html      |  7 +++-
 doc/html/man/form_field_opts.3x.html  | 14 +++++--
 doc/html/man/form_new_page.3x.html    |  9 ++---
 doc/html/man/infocmp.1m.html          |  4 +-
 doc/html/man/infotocap.1m.html        |  2 +-
 doc/html/man/menu.3x.html             |  2 +-
 doc/html/man/ncurses.3x.html          |  2 +-
 doc/html/man/ncurses6-config.1.html   |  2 +-
 doc/html/man/panel.3x.html            |  2 +-
 doc/html/man/scr_dump.5.html          |  6 +--
 doc/html/man/tabs.1.html              |  8 ++--
 doc/html/man/term.5.html              |  2 +-
 doc/html/man/terminfo.5.html          |  8 ++--
 doc/html/man/tic.1m.html              |  8 ++--
 doc/html/man/toe.1m.html              |  2 +-
 doc/html/man/tput.1.html              |  6 +--
 doc/html/man/tset.1.html              |  6 +--
 man/clear.1                           |  7 ++--
 man/curs_addch.3x                     |  7 ++--
 man/infocmp.1m                        | 31 +++++++++-------
 man/scr_dump.5                        | 18 +++++----
 man/tabs.1                            | 19 ++++++----
 man/term.5                            |  8 ++--
 man/terminfo.head                     | 11 +++---
 man/terminfo.tail                     | 52 +++++++++++++-------------
 man/tic.1m                            | 17 +++++----
 man/tput.1                            | 22 +++++++----
 man/tset.1                            | 10 +++--
 misc/terminfo.src                     | 53 ++++++++++++++++-----------
 ncurses/base/lib_color.c              |  9 +++--
 package/debian-mingw/changelog        |  4 +-
 package/debian-mingw64/changelog      |  4 +-
 package/debian/changelog              |  4 +-
 package/mingw-ncurses.nsi             |  4 +-
 package/mingw-ncurses.spec            |  2 +-
 package/ncurses.spec                  |  2 +-
 44 files changed, 231 insertions(+), 176 deletions(-)

diff --git a/NEWS b/NEWS
index da3b1d26..f01ae31e 100644
--- a/NEWS
+++ b/NEWS
@@ -25,7 +25,7 @@
 -- sale, use or other dealings in this Software without prior written        --
 -- authorization.                                                            --
 -------------------------------------------------------------------------------
--- $Id: NEWS,v 1.3127 2018/05/12 23:36:35 tom Exp $
+-- $Id: NEWS,v 1.3131 2018/05/19 21:17:03 tom Exp $
 -------------------------------------------------------------------------------
 
 This is a log of changes that ncurses has gone through since Zeyd started
@@ -45,6 +45,15 @@ See the AUTHORS file for the corresponding full names.
 Changes through 1.9.9e did not credit all contributions;
 it is not possible to add this information.
 
+20180519
+	+ formatting fixes for manpages, regenerate man-html documentation.
+	+ trim spurious whitespace from tmux in 2018-02-24 changes;
+	  fix some inconsistencies in/between tmux- and iterm2-entries for SGR
+	  (report by C Anthony Risinger)
+	+ improve iterm2 using some xterm features which it has adapted -TD
+	+ add check in pair_content() to handle the case where caller asks
+	  for an uninitialized pair (Debian #898658).
+
 20180512
 	+ remove trailing ';' from GCC_DEPRECATED definition.
 	+ repair a change from 20110730 which left an error-check/warning dead.
diff --git a/VERSION b/VERSION
index fc22197b..ea756714 100644
--- a/VERSION
+++ b/VERSION
@@ -1 +1 @@
-5:0:10	6.1	20180512
+5:0:10	6.1	20180519
diff --git a/dist.mk b/dist.mk
index 8752e197..a09a0f90 100644
--- a/dist.mk
+++ b/dist.mk
@@ -25,7 +25,7 @@
 # use or other dealings in this Software without prior written               #
 # authorization.                                                             #
 ##############################################################################
-# $Id: dist.mk,v 1.1223 2018/05/12 10:58:29 tom Exp $
+# $Id: dist.mk,v 1.1224 2018/05/16 00:44:03 tom Exp $
 # Makefile for creating ncurses distributions.
 #
 # This only needs to be used directly as a makefile by developers, but
@@ -37,7 +37,7 @@ SHELL = /bin/sh
 # These define the major/minor/patch versions of ncurses.
 NCURSES_MAJOR = 6
 NCURSES_MINOR = 1
-NCURSES_PATCH = 20180512
+NCURSES_PATCH = 20180519
 
 # We don't append the patch to the version, since this only applies to releases
 VERSION = $(NCURSES_MAJOR).$(NCURSES_MINOR)
diff --git a/doc/html/man/adacurses6-config.1.html b/doc/html/man/adacurses6-config.1.html
index bf3c2765..aa51c701 100644
--- a/doc/html/man/adacurses6-config.1.html
+++ b/doc/html/man/adacurses6-config.1.html
@@ -125,7 +125,7 @@ <H1 class="no-header">ADACURSES 1   User Commands</H1>
 </PRE><H2><a name="h2-SEE-ALSO">SEE ALSO</a></H2><PRE>
        <STRONG><A HREF="ncurses.3x.html">curses(3x)</A></STRONG>
 
-       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180407).
+       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180519).
 
 
 
diff --git a/doc/html/man/captoinfo.1m.html b/doc/html/man/captoinfo.1m.html
index 1b7e97a2..980a8c3e 100644
--- a/doc/html/man/captoinfo.1m.html
+++ b/doc/html/man/captoinfo.1m.html
@@ -190,7 +190,7 @@ <H1 class="no-header">captoinfo 1m</H1>
 </PRE><H2><a name="h2-SEE-ALSO">SEE ALSO</a></H2><PRE>
        <STRONG><A HREF="infocmp.1m.html">infocmp(1m)</A></STRONG>, <STRONG><A HREF="ncurses.3x.html">curses(3x)</A></STRONG>, <STRONG><A HREF="terminfo.5.html">terminfo(5)</A></STRONG>
 
-       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180407).
+       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180519).
 
 
 </PRE><H2><a name="h2-AUTHOR">AUTHOR</a></H2><PRE>
diff --git a/doc/html/man/clear.1.html b/doc/html/man/clear.1.html
index 54a1c753..a984574b 100644
--- a/doc/html/man/clear.1.html
+++ b/doc/html/man/clear.1.html
@@ -1,6 +1,6 @@
 <!-- 
   ****************************************************************************
-  * Copyright (c) 1998-2016,2017 Free Software Foundation, Inc.              *
+  * Copyright (c) 1998-2017,2018 Free Software Foundation, Inc.              *
   *                                                                          *
   * Permission is hereby granted, free of charge, to any person obtaining a  *
   * copy of this software and associated documentation files (the            *
@@ -26,7 +26,7 @@
   * sale, use or other dealings in this Software without prior written       *
   * authorization.                                                           *
   ****************************************************************************
-  * @Id: clear.1,v 1.19 2017/11/18 23:51:59 tom Exp @
+  * @Id: clear.1,v 1.20 2018/05/19 21:03:03 tom Exp @
   * these would be fallbacks for DS/DE,
   * but groff changed the meaning of the macros.
 -->
@@ -148,7 +148,7 @@ <H1 class="no-header">clear 1</H1>
 </PRE><H2><a name="h2-SEE-ALSO">SEE ALSO</a></H2><PRE>
        <STRONG><A HREF="tput.1.html">tput(1)</A></STRONG>, <STRONG><A HREF="terminfo.5.html">terminfo(5)</A></STRONG>
 
-       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180407).
+       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180519).
 
 
 
diff --git a/doc/html/man/curs_addch.3x.html b/doc/html/man/curs_addch.3x.html
index 290d7226..0202da63 100644
--- a/doc/html/man/curs_addch.3x.html
+++ b/doc/html/man/curs_addch.3x.html
@@ -1,7 +1,7 @@
 <!-- 
   * t
   ****************************************************************************
-  * Copyright (c) 1998-2015,2017 Free Software Foundation, Inc.              *
+  * Copyright (c) 1998-2017,2018 Free Software Foundation, Inc.              *
   *                                                                          *
   * Permission is hereby granted, free of charge, to any person obtaining a  *
   * copy of this software and associated documentation files (the            *
@@ -27,7 +27,7 @@
   * sale, use or other dealings in this Software without prior written       *
   * authorization.                                                           *
   ****************************************************************************
-  * @Id: curs_addch.3x,v 1.44 2017/11/20 01:27:20 tom Exp @
+  * @Id: curs_addch.3x,v 1.45 2018/05/19 21:03:18 tom Exp @
 -->
 <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
 <HTML>
diff --git a/doc/html/man/form.3x.html b/doc/html/man/form.3x.html
index 94384c71..ea156485 100644
--- a/doc/html/man/form.3x.html
+++ b/doc/html/man/form.3x.html
@@ -246,7 +246,7 @@ <H1 class="no-header">form 3x</H1>
        <STRONG><A HREF="ncurses.3x.html">curses(3x)</A></STRONG>  and  related  pages  whose names begin "form_" for detailed
        descriptions of the entry points.
 
-       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180407).
+       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180519).
 
 
 
diff --git a/doc/html/man/form_driver.3x.html b/doc/html/man/form_driver.3x.html
index 01824206..b282953e 100644
--- a/doc/html/man/form_driver.3x.html
+++ b/doc/html/man/form_driver.3x.html
@@ -1,6 +1,6 @@
 <!-- 
   ****************************************************************************
-  * Copyright (c) 1998-2016,2017 Free Software Foundation, Inc.              *
+  * Copyright (c) 1998-2017,2018 Free Software Foundation, Inc.              *
   *                                                                          *
   * Permission is hereby granted, free of charge, to any person obtaining a  *
   * copy of this software and associated documentation files (the            *
@@ -26,7 +26,7 @@
   * sale, use or other dealings in this Software without prior written       *
   * authorization.                                                           *
   ****************************************************************************
-  * @Id: form_driver.3x,v 1.28 2017/11/18 23:47:37 tom Exp @
+  * @Id: form_driver.3x,v 1.29 2018/04/28 19:58:58 tom Exp @
 -->
 <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
 <HTML>
@@ -243,6 +243,9 @@ <H1 class="no-header">form_driver 3x</H1>
        <STRONG>E_INVALID_FIELD</STRONG>
             Contents of field is invalid.
 
+       <STRONG>E_NOT_CONNECTED</STRONG>
+            No fields are connected to the form.
+
        <STRONG>E_REQUEST_DENIED</STRONG>
             The form driver could not process the request.
 
diff --git a/doc/html/man/form_field_opts.3x.html b/doc/html/man/form_field_opts.3x.html
index db1e41f7..32b78585 100644
--- a/doc/html/man/form_field_opts.3x.html
+++ b/doc/html/man/form_field_opts.3x.html
@@ -1,7 +1,7 @@
 <!-- 
   * t
   ****************************************************************************
-  * Copyright (c) 1998-2014,2015 Free Software Foundation, Inc.              *
+  * Copyright (c) 1998-2015,2018 Free Software Foundation, Inc.              *
   *                                                                          *
   * Permission is hereby granted, free of charge, to any person obtaining a  *
   * copy of this software and associated documentation files (the            *
@@ -27,7 +27,7 @@
   * sale, use or other dealings in this Software without prior written       *
   * authorization.                                                           *
   ****************************************************************************
-  * @Id: form_field_opts.3x,v 1.20 2015/12/05 23:53:43 tom Exp @
+  * @Id: form_field_opts.3x,v 1.21 2018/04/28 21:24:57 tom Exp @
 -->
 <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
 <HTML>
@@ -118,6 +118,14 @@ <H1 class="no-header">form_field_opts 3x</H1>
             Preserve leading whitespace in the field buffer, which is normally
             discarded.
 
+       O_EDGE_INSERT_STAY
+            When inserting into a field up to the boundary  position,  option-
+            ally  delay  the  scrolling,  so  that the last inserted character
+            remains visible, but advance the cursor to reflect the  insertion.
+            This  allows the form library to display the inserted character in
+            one-character fields as well as allowing the library  to  maintain
+            consistent state.
+
 
 </PRE><H2><a name="h2-RETURN-VALUE">RETURN VALUE</a></H2><PRE>
        Except for <STRONG>field_opts</STRONG>, each routine returns one of the following:
@@ -144,7 +152,7 @@ <H1 class="no-header">form_field_opts 3x</H1>
 
 
 </PRE><H2><a name="h2-PORTABILITY">PORTABILITY</a></H2><PRE>
-       These  routines emulate the System V forms library.  They were not sup-
+       These routines emulate the System V forms library.  They were not  sup-
        ported on Version 7 or BSD versions.
 
 
diff --git a/doc/html/man/form_new_page.3x.html b/doc/html/man/form_new_page.3x.html
index a68bb70d..b02c0363 100644
--- a/doc/html/man/form_new_page.3x.html
+++ b/doc/html/man/form_new_page.3x.html
@@ -1,7 +1,7 @@
 <!-- 
   * t
   ****************************************************************************
-  * Copyright (c) 1998-2010,2015 Free Software Foundation, Inc.              *
+  * Copyright (c) 1998-2015,2018 Free Software Foundation, Inc.              *
   *                                                                          *
   * Permission is hereby granted, free of charge, to any person obtaining a  *
   * copy of this software and associated documentation files (the            *
@@ -27,7 +27,7 @@
   * sale, use or other dealings in this Software without prior written       *
   * authorization.                                                           *
   ****************************************************************************
-  * @Id: form_new_page.3x,v 1.11 2015/12/05 23:42:45 tom Exp @
+  * @Id: form_new_page.3x,v 1.12 2018/04/28 19:58:50 tom Exp @
 -->
 <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
 <HTML>
@@ -67,13 +67,10 @@ <H1 class="no-header">form_new_page 3x</H1>
 </PRE><H2><a name="h2-RETURN-VALUE">RETURN VALUE</a></H2><PRE>
        The function <STRONG>new_page</STRONG> returns <STRONG>TRUE</STRONG> or <STRONG>FALSE</STRONG>.
 
-       The function <STRONG>set_new_page</STRONG> return one of the following:
+       The function <STRONG>set_new_page</STRONG> returns one of the following:
 
        <STRONG>E_OK</STRONG> The routine succeeded.
 
-       <STRONG>E_SYSTEM_ERROR</STRONG>
-            System error occurred (see <STRONG>errno</STRONG>).
-
        <STRONG>E_CONNECTED</STRONG>
             The given field is already connected to a form.
 
diff --git a/doc/html/man/infocmp.1m.html b/doc/html/man/infocmp.1m.html
index 8dddb622..851a2c13 100644
--- a/doc/html/man/infocmp.1m.html
+++ b/doc/html/man/infocmp.1m.html
@@ -27,7 +27,7 @@
   * sale, use or other dealings in this Software without prior written       *
   * authorization.                                                           *
   ****************************************************************************
-  * @Id: infocmp.1m,v 1.70 2018/01/23 10:10:17 tom Exp @
+  * @Id: infocmp.1m,v 1.71 2018/05/19 21:07:17 tom Exp @
 -->
 <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
 <HTML>
@@ -481,7 +481,7 @@ <H1 class="no-header">infocmp 1m</H1>
 
        https://invisible-island.net/ncurses/tctest.html
 
-       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180407).
+       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180519).
 
 
 </PRE><H2><a name="h2-AUTHOR">AUTHOR</a></H2><PRE>
diff --git a/doc/html/man/infotocap.1m.html b/doc/html/man/infotocap.1m.html
index c43169f8..36ee54cc 100644
--- a/doc/html/man/infotocap.1m.html
+++ b/doc/html/man/infotocap.1m.html
@@ -85,7 +85,7 @@ <H1 class="no-header">infotocap 1m</H1>
 </PRE><H2><a name="h2-SEE-ALSO">SEE ALSO</a></H2><PRE>
        <STRONG><A HREF="ncurses.3x.html">curses(3x)</A></STRONG>, <STRONG><A HREF="tic.1m.html">tic(1m)</A></STRONG>, <STRONG><A HREF="infocmp.1m.html">infocmp(1m)</A></STRONG>, <STRONG><A HREF="terminfo.5.html">terminfo(5)</A></STRONG>
 
-       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180407).
+       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180519).
 
 
 </PRE><H2><a name="h2-AUTHOR">AUTHOR</a></H2><PRE>
diff --git a/doc/html/man/menu.3x.html b/doc/html/man/menu.3x.html
index 9af6758e..65d749de 100644
--- a/doc/html/man/menu.3x.html
+++ b/doc/html/man/menu.3x.html
@@ -221,7 +221,7 @@ <H1 class="no-header">menu 3x</H1>
        <STRONG><A HREF="ncurses.3x.html">curses(3x)</A></STRONG>  and  related  pages  whose names begin "menu_" for detailed
        descriptions of the entry points.
 
-       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180407).
+       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180519).
 
 
 
diff --git a/doc/html/man/ncurses.3x.html b/doc/html/man/ncurses.3x.html
index 697226b7..57dc5259 100644
--- a/doc/html/man/ncurses.3x.html
+++ b/doc/html/man/ncurses.3x.html
@@ -59,7 +59,7 @@ <H1 class="no-header">ncurses 3x</H1>
        method of updating  character  screens  with  reasonable  optimization.
        This  implementation  is  "new  curses"  (ncurses)  and is the approved
        replacement for 4.4BSD classic curses,  which  has  been  discontinued.
-       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180407).
+       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180519).
 
        The  <STRONG>ncurses</STRONG>  library emulates the curses library of System V Release 4
        UNIX, and XPG4 (X/Open Portability Guide) curses  (also  known  as  XSI
diff --git a/doc/html/man/ncurses6-config.1.html b/doc/html/man/ncurses6-config.1.html
index b8677eb2..4e7a8bf0 100644
--- a/doc/html/man/ncurses6-config.1.html
+++ b/doc/html/man/ncurses6-config.1.html
@@ -112,7 +112,7 @@ <H1 class="no-header">ncurses6-config 1</H1>
 </PRE><H2><a name="h2-SEE-ALSO">SEE ALSO</a></H2><PRE>
        <STRONG><A HREF="ncurses.3x.html">curses(3x)</A></STRONG>
 
-       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180407).
+       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180519).
 
 
 
diff --git a/doc/html/man/panel.3x.html b/doc/html/man/panel.3x.html
index 25edaa35..a8ba3393 100644
--- a/doc/html/man/panel.3x.html
+++ b/doc/html/man/panel.3x.html
@@ -204,7 +204,7 @@ <H1 class="no-header">panel 3x</H1>
 </PRE><H2><a name="h2-SEE-ALSO">SEE ALSO</a></H2><PRE>
        <STRONG><A HREF="ncurses.3x.html">curses(3x)</A></STRONG>, <STRONG><A HREF="curs_variables.3x.html">curs_variables(3x)</A></STRONG>,
 
-       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180407).
+       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180519).
 
 
 </PRE><H2><a name="h2-AUTHOR">AUTHOR</a></H2><PRE>
diff --git a/doc/html/man/scr_dump.5.html b/doc/html/man/scr_dump.5.html
index 081c90ed..206c1380 100644
--- a/doc/html/man/scr_dump.5.html
+++ b/doc/html/man/scr_dump.5.html
@@ -1,6 +1,6 @@
 <!-- 
   ****************************************************************************
-  * Copyright (c) 2017 Free Software Foundation, Inc.                        *
+  * Copyright (c) 2017,2018 Free Software Foundation, Inc.                   *
   *                                                                          *
   * Permission is hereby granted, free of charge, to any person obtaining a  *
   * copy of this software and associated documentation files (the            *
@@ -26,7 +26,7 @@
   * sale, use or other dealings in this Software without prior written       *
   * authorization.                                                           *
   ****************************************************************************
-  * @Id: scr_dump.5,v 1.12 2017/11/25 20:12:55 tom Exp @
+  * @Id: scr_dump.5,v 1.13 2018/05/19 21:10:21 tom Exp @
 -->
 <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
 <HTML>
@@ -154,7 +154,7 @@ <H1 class="no-header">scr_dump 5</H1>
           <STRONG>o</STRONG>   Shading is used to identify  <EM>X/Open</EM>  <EM>Enhanced</EM>  <EM>Curses</EM>  material,
               relating to interfaces included to provide enhanced capabilities
               for applications originally written to be  compiled  on  systems
-              based  on  the  UNIX  operating  system. Therefore, the features
+              based  on  the  UNIX  operating system.  Therefore, the features
               described may not be present on systems that conform to <STRONG>XPG4</STRONG>  <STRONG>or</STRONG>
               <STRONG>to</STRONG>  <STRONG>earlier</STRONG> <STRONG>XPG</STRONG> <STRONG>releases</STRONG>.  The relevant reference pages may pro-
               vide additional or more specific portability warnings about  use
diff --git a/doc/html/man/tabs.1.html b/doc/html/man/tabs.1.html
index eef270d8..b3ba336c 100644
--- a/doc/html/man/tabs.1.html
+++ b/doc/html/man/tabs.1.html
@@ -1,6 +1,6 @@
 <!-- 
   ****************************************************************************
-  * Copyright (c) 2008-2016,2017 Free Software Foundation, Inc.              *
+  * Copyright (c) 2008-2017,2018 Free Software Foundation, Inc.              *
   *                                                                          *
   * Permission is hereby granted, free of charge, to any person obtaining a  *
   * copy of this software and associated documentation files (the            *
@@ -26,7 +26,7 @@
   * sale, use or other dealings in this Software without prior written       *
   * authorization.                                                           *
   ****************************************************************************
-  * @Id: tabs.1,v 1.16 2017/11/18 23:56:00 tom Exp @
+  * @Id: tabs.1,v 1.17 2018/05/19 21:07:46 tom Exp @
 -->
 <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
 <HTML>
@@ -141,7 +141,7 @@ <H1 class="no-header">tabs 1</H1>
 
 
 </PRE><H2><a name="h2-PORTABILITY">PORTABILITY</a></H2><PRE>
-       IEEE  Std  1003.1/The  Open  Group    Base   Specifications   Issue   7
+       <EM>IEEE</EM>  <EM>Std</EM>  <EM>1003.1/The</EM>  <EM>Open</EM>   <EM>Group</EM>   <EM>Base</EM>   <EM>Specifications</EM>   <EM>Issue</EM>   <EM>7</EM>
        (POSIX.1-2008) describes a <STRONG>tabs</STRONG> utility.  However
 
        <STRONG>o</STRONG>   This  standard describes a <STRONG>+m</STRONG> option, to set a terminal's left-mar-
@@ -164,7 +164,7 @@ <H1 class="no-header">tabs 1</H1>
 </PRE><H2><a name="h2-SEE-ALSO">SEE ALSO</a></H2><PRE>
        <STRONG><A HREF="tset.1.html">tset(1)</A></STRONG>, <STRONG><A HREF="infocmp.1m.html">infocmp(1m)</A></STRONG>, <STRONG><A HREF="ncurses.3x.html">curses(3x)</A></STRONG>, <STRONG><A HREF="terminfo.5.html">terminfo(5)</A></STRONG>.
 
-       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180407).
+       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180519).
 
 
 
diff --git a/doc/html/man/term.5.html b/doc/html/man/term.5.html
index 7edfa1dc..3337068a 100644
--- a/doc/html/man/term.5.html
+++ b/doc/html/man/term.5.html
@@ -26,7 +26,7 @@
   * sale, use or other dealings in this Software without prior written       *
   * authorization.                                                           *
   ****************************************************************************
-  * @Id: term.5,v 1.28 2018/03/31 22:41:29 tom Exp @
+  * @Id: term.5,v 1.29 2018/05/19 21:09:25 tom Exp @
 -->
 <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
 <HTML>
diff --git a/doc/html/man/terminfo.5.html b/doc/html/man/terminfo.5.html
index 763ad5f0..c3b2183d 100644
--- a/doc/html/man/terminfo.5.html
+++ b/doc/html/man/terminfo.5.html
@@ -31,9 +31,9 @@
   * sale, use or other dealings in this Software without prior written       *
   * authorization.                                                           *
   ****************************************************************************
-  * @Id: terminfo.head,v 1.32 2017/04/22 13:52:49 tom Exp @
+  * @Id: terminfo.head,v 1.33 2018/05/19 20:57:27 tom Exp @
   * Head of terminfo man page ends here
-  * @Id: terminfo.tail,v 1.88 2017/08/12 22:26:02 tom Exp @
+  * @Id: terminfo.tail,v 1.89 2018/05/19 21:01:52 tom Exp @
   * Beginning of terminfo.tail file
   * This file is part of ncurses.
   * See "terminfo.head" for copyright.
@@ -74,7 +74,7 @@ <H1 class="no-header">terminfo 5   File Formats</H1>
        <EM>Terminfo</EM> describes terminals by giving a set of capabilities which they
        have, by specifying how to perform screen operations, and by specifying
        padding  requirements  and  initialization  sequences.   This describes
-       <STRONG>ncurses</STRONG> version 6.1 (patch 20180407).
+       <STRONG>ncurses</STRONG> version 6.1 (patch 20180519).
 
 
 </PRE><H3><a name="h3-Terminfo-Entry-Syntax">Terminfo Entry Syntax</a></H3><PRE>
@@ -142,7 +142,7 @@ <H1 class="no-header">terminfo 5   File Formats</H1>
             -vb      Use visible bell instead of beep         wy370-vb
             -w       Wide mode (&gt; 80 columns, usually 132)    vt100-w
 
-       For more on terminal naming conventions, see the <STRONG>term(7)</STRONG> manual page.
+       For more on terminal naming conventions, see the <STRONG><A HREF="term.7.html">term(7)</A></STRONG> manual page.
 
 
 </PRE><H3><a name="h3-Terminfo-Capabilities-Syntax">Terminfo Capabilities Syntax</a></H3><PRE>
diff --git a/doc/html/man/tic.1m.html b/doc/html/man/tic.1m.html
index d50290c7..9dfe7197 100644
--- a/doc/html/man/tic.1m.html
+++ b/doc/html/man/tic.1m.html
@@ -1,6 +1,6 @@
 <!-- 
   ****************************************************************************
-  * Copyright (c) 1998-2016,2017 Free Software Foundation, Inc.              *
+  * Copyright (c) 1998-2017,2018 Free Software Foundation, Inc.              *
   *                                                                          *
   * Permission is hereby granted, free of charge, to any person obtaining a  *
   * copy of this software and associated documentation files (the            *
@@ -26,7 +26,7 @@
   * sale, use or other dealings in this Software without prior written       *
   * authorization.                                                           *
   ****************************************************************************
-  * @Id: tic.1m,v 1.66 2017/11/18 23:51:17 tom Exp @
+  * @Id: tic.1m,v 1.67 2018/05/19 21:07:46 tom Exp @
 -->
 <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
 <HTML>
@@ -140,7 +140,7 @@ <H1 class="no-header">tic 1m</H1>
                   the end of the string will not convert completely.
 
        <STRONG>-c</STRONG>     tells <STRONG>tic</STRONG> to only check <EM>file</EM> for errors, including syntax  prob-
-              lems  and  bad  use  links.   If  you  specify <STRONG>-C</STRONG> (<STRONG>-I</STRONG>) with this
+              lems  and  bad  use-links.   If  you  specify  <STRONG>-C</STRONG> (<STRONG>-I</STRONG>) with this
               option, the code will print warnings about entries which,  after
               use  resolution, are more than 1023 (4096) bytes long.  Due to a
               fixed buffer length in older termcap libraries, as well as buggy
@@ -364,7 +364,7 @@ <H1 class="no-header">tic 1m</H1>
        <STRONG><A HREF="infocmp.1m.html">infocmp(1m)</A></STRONG>,   <STRONG><A HREF="captoinfo.1m.html">captoinfo(1m)</A></STRONG>,   <STRONG><A HREF="infotocap.1m.html">infotocap(1m)</A></STRONG>,   <STRONG><A HREF="toe.1m.html">toe(1m)</A></STRONG>,   <STRONG><A HREF="ncurses.3x.html">curses(3x)</A></STRONG>,
        <STRONG><A HREF="term.5.html">term(5)</A></STRONG>.  <STRONG><A HREF="terminfo.5.html">terminfo(5)</A></STRONG>.
 
-       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180407).
+       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180519).
 
 
 </PRE><H2><a name="h2-AUTHOR">AUTHOR</a></H2><PRE>
diff --git a/doc/html/man/toe.1m.html b/doc/html/man/toe.1m.html
index cc5e66c5..b85685e2 100644
--- a/doc/html/man/toe.1m.html
+++ b/doc/html/man/toe.1m.html
@@ -113,7 +113,7 @@ <H1 class="no-header">toe 1m</H1>
        <STRONG><A HREF="tic.1m.html">tic(1m)</A></STRONG>, <STRONG><A HREF="infocmp.1m.html">infocmp(1m)</A></STRONG>, <STRONG><A HREF="captoinfo.1m.html">captoinfo(1m)</A></STRONG>,  <STRONG><A HREF="infotocap.1m.html">infotocap(1m)</A></STRONG>,  <STRONG><A HREF="ncurses.3x.html">curses(3x)</A></STRONG>,  <STRONG>ter-</STRONG>
        <STRONG><A HREF="terminfo.5.html">minfo(5)</A></STRONG>.
 
-       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180407).
+       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180519).
 
 
 
diff --git a/doc/html/man/tput.1.html b/doc/html/man/tput.1.html
index 20678991..4b6ec0ff 100644
--- a/doc/html/man/tput.1.html
+++ b/doc/html/man/tput.1.html
@@ -1,7 +1,7 @@
 <!-- 
   * t
   ****************************************************************************
-  * Copyright (c) 1998-2016,2017 Free Software Foundation, Inc.              *
+  * Copyright (c) 1998-2017,2018 Free Software Foundation, Inc.              *
   *                                                                          *
   * Permission is hereby granted, free of charge, to any person obtaining a  *
   * copy of this software and associated documentation files (the            *
@@ -27,7 +27,7 @@
   * sale, use or other dealings in this Software without prior written       *
   * authorization.                                                           *
   ****************************************************************************
-  * @Id: tput.1,v 1.57 2017/11/20 01:07:02 tom Exp @
+  * @Id: tput.1,v 1.58 2018/05/19 21:07:46 tom Exp @
 -->
 <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
 <HTML>
@@ -482,7 +482,7 @@ <H1 class="no-header">tput 1</H1>
 </PRE><H2><a name="h2-SEE-ALSO">SEE ALSO</a></H2><PRE>
        <STRONG><A HREF="clear.1.html">clear(1)</A></STRONG>, <STRONG>stty(1)</STRONG>, <STRONG><A HREF="tabs.1.html">tabs(1)</A></STRONG>, <STRONG><A HREF="tset.1.html">tset(1)</A></STRONG>, <STRONG><A HREF="terminfo.5.html">terminfo(5)</A></STRONG>, <STRONG><A HREF="curs_termcap.3x.html">curs_termcap(3x)</A></STRONG>.
 
-       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180407).
+       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180519).
 
 
 
diff --git a/doc/html/man/tset.1.html b/doc/html/man/tset.1.html
index 0dfc28c7..ad94c427 100644
--- a/doc/html/man/tset.1.html
+++ b/doc/html/man/tset.1.html
@@ -1,6 +1,6 @@
 <!-- 
   ****************************************************************************
-  * Copyright (c) 1998-2016,2017 Free Software Foundation, Inc.              *
+  * Copyright (c) 1998-2017,2018 Free Software Foundation, Inc.              *
   *                                                                          *
   * Permission is hereby granted, free of charge, to any person obtaining a  *
   * copy of this software and associated documentation files (the            *
@@ -26,7 +26,7 @@
   * sale, use or other dealings in this Software without prior written       *
   * authorization.                                                           *
   ****************************************************************************
-  * @Id: tset.1,v 1.52 2017/11/20 01:27:53 tom Exp @
+  * @Id: tset.1,v 1.53 2018/05/19 21:07:46 tom Exp @
 -->
 <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
 <HTML>
@@ -389,7 +389,7 @@ <H1 class="no-header">tset 1</H1>
        <STRONG>csh(1)</STRONG>,  <STRONG>sh(1)</STRONG>,  <STRONG>stty(1)</STRONG>,   <STRONG><A HREF="curs_terminfo.3x.html">curs_terminfo(3x)</A></STRONG>,   <STRONG>tty(4)</STRONG>,   <STRONG><A HREF="terminfo.5.html">terminfo(5)</A></STRONG>,
        <STRONG>ttys(5)</STRONG>, <STRONG>environ(7)</STRONG>
 
-       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180407).
+       This describes <STRONG>ncurses</STRONG> version 6.1 (patch 20180519).
 
 
 
diff --git a/man/clear.1 b/man/clear.1
index 2c4fb22a..6b611da2 100644
--- a/man/clear.1
+++ b/man/clear.1
@@ -1,5 +1,5 @@
 .\"***************************************************************************
-.\" Copyright (c) 1998-2016,2017 Free Software Foundation, Inc.              *
+.\" Copyright (c) 1998-2017,2018 Free Software Foundation, Inc.              *
 .\"                                                                          *
 .\" Permission is hereby granted, free of charge, to any person obtaining a  *
 .\" copy of this software and associated documentation files (the            *
@@ -26,7 +26,7 @@
 .\" authorization.                                                           *
 .\"***************************************************************************
 .\"
-.\" $Id: clear.1,v 1.19 2017/11/18 23:51:59 tom Exp $
+.\" $Id: clear.1,v 1.20 2018/05/19 21:03:03 tom Exp $
 .TH @CLEAR@ 1 ""
 .\" these would be fallbacks for DS/DE,
 .\" but groff changed the meaning of the macros.
@@ -59,7 +59,8 @@
 .br
 .SH DESCRIPTION
 \fB@CLEAR@\fR clears your screen if this is possible,
-including its scrollback buffer (if the extended \*(``E3\*('' capability is defined).
+including its scrollback buffer
+(if the extended \*(``E3\*('' capability is defined).
 \fB@CLEAR@\fR looks in the environment for the terminal type
 given by the environment variable \fBTERM\fP,
 and then in the
diff --git a/man/curs_addch.3x b/man/curs_addch.3x
index cfcb4a26..664b8da5 100644
--- a/man/curs_addch.3x
+++ b/man/curs_addch.3x
@@ -1,6 +1,6 @@
 '\" t
 .\"***************************************************************************
-.\" Copyright (c) 1998-2015,2017 Free Software Foundation, Inc.              *
+.\" Copyright (c) 1998-2017,2018 Free Software Foundation, Inc.              *
 .\"                                                                          *
 .\" Permission is hereby granted, free of charge, to any person obtaining a  *
 .\" copy of this software and associated documentation files (the            *
@@ -27,7 +27,7 @@
 .\" authorization.                                                           *
 .\"***************************************************************************
 .\"
-.\" $Id: curs_addch.3x,v 1.44 2017/11/20 01:27:20 tom Exp $
+.\" $Id: curs_addch.3x,v 1.45 2018/05/19 21:03:18 tom Exp $
 .TH curs_addch 3X ""
 .ie \n(.g .ds `` \(lq
 .el       .ds `` ``
@@ -63,7 +63,8 @@
 .SS Adding characters
 The \fBaddch\fR, \fBwaddch\fR, \fBmvaddch\fR and \fBmvwaddch\fR routines put
 the character \fIch\fR into the given window at its current window position,
-which is then advanced.  They are analogous to \fBputchar\fR(3) in \fBstdio\fR(3).
+which is then advanced.
+They are analogous to \fBputchar\fR(3) in \fBstdio\fR(3).
 If the advance is at the right margin:
 .bP
 The cursor automatically wraps to the beginning of the next line.
diff --git a/man/infocmp.1m b/man/infocmp.1m
index 34efbf63..19852e8f 100644
--- a/man/infocmp.1m
+++ b/man/infocmp.1m
@@ -27,7 +27,7 @@
 .\" authorization.                                                           *
 .\"***************************************************************************
 .\"
-.\" $Id: infocmp.1m,v 1.70 2018/01/23 10:10:17 tom Exp $
+.\" $Id: infocmp.1m,v 1.71 2018/05/19 21:07:17 tom Exp $
 .TH @INFOCMP@ 1M ""
 .ie \n(.g .ds `` \(lq
 .el       .ds `` ``
@@ -207,9 +207,9 @@ Mandatory padding is not supported.
 Because
 \fBtermcap\fR strings are not as flexible, it is not always possible to convert
 a \fBterminfo\fR string capability into an equivalent \fBtermcap\fR format.
-A subsequent conversion of the \fBtermcap\fR file back into \fBterminfo\fR format
-will not necessarily reproduce the original \fBterminfo\fR
-source.
+A subsequent conversion of the \fBtermcap\fR file
+back into \fBterminfo\fR format
+will not necessarily reproduce the original \fBterminfo\fR source.
 .PP
 Some common \fBterminfo\fR parameter sequences, their \fBtermcap\fR
 equivalents, and some terminal types which commonly have such sequences, are:
@@ -239,7 +239,8 @@ In this manner, it is possible to retrofit generic terminfo
 entries into a terminal's description.
 Or, if two similar terminals exist, but
 were coded at different times or by different people so that each description
-is a full description, using \fB@INFOCMP@\fR will show what can be done to change
+is a full description, using \fB@INFOCMP@\fR
+will show what can be done to change
 one description to be relative to the other.
 .PP
 A capability will get printed with an at-sign (@) if it no longer exists in the
@@ -303,12 +304,13 @@ the fields will be printed several to a line to a maximum width
 of 60 characters.
 .TP
 \fB\-a\fR
-tells \fB@INFOCMP@\fP to retain commented-out capabilities rather than discarding
-them.
+tells \fB@INFOCMP@\fP to retain commented-out capabilities
+rather than discarding them.
 Capabilities are commented by prefixing them with a period.
 .TP
 \fB\-D\fR
-tells \fB@INFOCMP@\fP to print the database locations that it knows about, and exit.
+tells \fB@INFOCMP@\fP to print the database locations that it knows about,
+and exit.
 .TP 5
 \fB\-E\fR
 Dump the capabilities of the given terminal as tables, needed in
@@ -461,7 +463,8 @@ This makes the output a little shorter:
 .RS
 .bP
 Make the comparison listing shorter by omitting subheadings, and using
-\*(``\-\*('' for absent capabilities, \*(``@\*('' for canceled rather than \*(``NULL\*(''.
+\*(``\-\*('' for absent capabilities, \*(``@\*(''
+for canceled rather than \*(``NULL\*(''.
 .bP
 However, show differences between absent and cancelled capabilities.
 .bP
@@ -477,8 +480,8 @@ that have their own extensions incompatible with SVr4/XSI.
 .RS
 .bP
 Available terminfo
-subsets are \*(``SVr1\*('', \*(``Ultrix\*('', \*(``HP\*('', and \*(``AIX\*(''; see \fBterminfo\fR(\*n) for
-details.
+subsets are \*(``SVr1\*('', \*(``Ultrix\*('', \*(``HP\*('', and \*(``AIX\*('';
+see \fBterminfo\fR(\*n) for details.
 .bP
 You can also choose the subset \*(``BSD\*('' which selects only capabilities
 with termcap equivalents recognized by 4.4BSD.
@@ -525,7 +528,8 @@ Normally when translating from terminfo to termcap,
 untranslatable capabilities are commented-out.
 .TP 5
 \fB\-U\fR
-tells \fB@INFOCMP@\fP to not post-process the data after parsing the source file.
+tells \fB@INFOCMP@\fP to not post-process the data
+after parsing the source file.
 This feature helps when comparing the actual contents of two source files,
 since it excludes the inferences that \fB@INFOCMP@\fP makes to fill in missing
 data.
@@ -583,7 +587,8 @@ Also, it shows missing integer capabilities as \fB\-1\fP
 This implementation shows those as \*(``NULL\*('',
 for consistency with missing strings.
 .PP
-The \fB\-r\fR option's notion of \*(``termcap\*('' capabilities is System V Release 4's.
+The \fB\-r\fR option's notion of \*(``termcap\*('' capabilities
+is System V Release 4's.
 Actual BSD curses versions will have a more restricted set.
 To see only the
 4.4BSD set, use \fB\-r\fR \fB\-RBSD\fR.
diff --git a/man/scr_dump.5 b/man/scr_dump.5
index b9e9bacd..bb2d1c7f 100644
--- a/man/scr_dump.5
+++ b/man/scr_dump.5
@@ -1,5 +1,5 @@
 .\"***************************************************************************
-.\" Copyright (c) 2017 Free Software Foundation, Inc.                        *
+.\" Copyright (c) 2017,2018 Free Software Foundation, Inc.                   *
 .\"                                                                          *
 .\" Permission is hereby granted, free of charge, to any person obtaining a  *
 .\" copy of this software and associated documentation files (the            *
@@ -26,7 +26,7 @@
 .\" authorization.                                                           *
 .\"***************************************************************************
 .\"
-.\" $Id: scr_dump.5,v 1.12 2017/11/25 20:12:55 tom Exp $
+.\" $Id: scr_dump.5,v 1.13 2018/05/19 21:10:21 tom Exp $
 .TH scr_dump 5
 .ie \n(.g .ds `` \(lq
 .el       .ds `` ``
@@ -155,12 +155,14 @@ because it did not meet the criteria for \fIbase curses\fP).
 The document explained the term \*(``enhanced\*('' as follows:
 .RS 3
 .bP
-Shading is used to identify \fIX/Open Enhanced Curses\fP material, relating to interfaces
-included to provide enhanced capabilities for applications originally written to be
-compiled on systems based on the UNIX operating system. Therefore, the features
-described may not be present on systems that conform to \fBXPG4 or to earlier XPG releases\fP.
-The relevant reference pages may provide additional or more specific portability warnings
-about use of the material.
+Shading is used to identify \fIX/Open Enhanced Curses\fP material,
+relating to interfaces included to provide enhanced capabilities
+for applications originally written to be compiled on systems
+based on the UNIX operating system.
+Therefore, the features described may not be present on systems
+that conform to \fBXPG4 or to earlier XPG releases\fP.
+The relevant reference pages may provide additional
+or more specific portability warnings about use of the material.
 .RE
 .PP
 In the foregoing, emphasis was added to \fBunspecified format\fP
diff --git a/man/tabs.1 b/man/tabs.1
index 6f973a61..ed3e41fa 100644
--- a/man/tabs.1
+++ b/man/tabs.1
@@ -1,5 +1,5 @@
 .\"***************************************************************************
-.\" Copyright (c) 2008-2016,2017 Free Software Foundation, Inc.              *
+.\" Copyright (c) 2008-2017,2018 Free Software Foundation, Inc.              *
 .\"                                                                          *
 .\" Permission is hereby granted, free of charge, to any person obtaining a  *
 .\" copy of this software and associated documentation files (the            *
@@ -26,7 +26,7 @@
 .\" authorization.                                                           *
 .\"***************************************************************************
 .\"
-.\" $Id: tabs.1,v 1.16 2017/11/18 23:56:00 tom Exp $
+.\" $Id: tabs.1,v 1.17 2018/05/19 21:07:46 tom Exp $
 .TH @TABS@ 1 ""
 .ds n 5
 .ie \n(.g .ds `` \(lq
@@ -93,9 +93,10 @@ The \fB@TABS@\fP program processes a single list of tab stops.
 The last option to be processed which defines a list is the one that
 determines the list to be processed.
 .SS Implicit Lists
-Use a single number as an option, e.g., \*(``\fB\-5\fP\*('' to set tabs at the given
-interval (in this case 1, 6, 11, 16, 21, etc.).  Tabs are repeated up to
-the right margin of the screen.
+Use a single number as an option,
+e.g., \*(``\fB\-5\fP\*('' to set tabs at the given
+interval (in this case 1, 6, 11, 16, 21, etc.).
+Tabs are repeated up to the right margin of the screen.
 .PP
 Use \*(``\fB\-0\fP\*('' to clear all tabs.
 .PP
@@ -112,7 +113,8 @@ tabs 1,6,11,16,21
 tabs 1 6 11 16 21
 .NE
 .PP
-Use a \*(``+\*('' to treat a number as an increment relative to the previous value,
+Use a \*(``+\*('' to treat a number
+as an increment relative to the previous value,
 e.g.,
 .NS
 tabs 1,+5,+5,+5,+5
@@ -150,8 +152,9 @@ SNOBOL
 UNIVAC 1100 Assembler
 .SH PORTABILITY
 .PP
-IEEE Std 1003.1/The Open Group  Base Specifications Issue 7 (POSIX.1-2008) describes
-a \fBtabs\fP utility.  However
+\fIIEEE Std 1003.1/The Open Group Base Specifications Issue 7\fP (POSIX.1-2008)
+describes a \fBtabs\fP utility.
+However
 .bP
 This standard describes a \fB+m\fP option, to set a terminal's left-margin.
 Very few of the entries in the terminal database provide this capability.
diff --git a/man/term.5 b/man/term.5
index 554e8ed5..59489037 100644
--- a/man/term.5
+++ b/man/term.5
@@ -26,7 +26,7 @@
 .\" authorization.                                                           *
 .\"***************************************************************************
 .\"
-.\" $Id: term.5,v 1.28 2018/03/31 22:41:29 tom Exp $
+.\" $Id: term.5,v 1.29 2018/05/19 21:09:25 tom Exp $
 .TH term 5
 .ie \n(.g .ds `` \(lq
 .el       .ds `` ``
@@ -180,7 +180,8 @@ With some minor variations of the offsets (see PORTABILITY),
 the same binary format is used in all modern UNIX systems.
 Each system uses a predefined set of boolean, number or string capabilities.
 .PP
-The \fBncurses\fP libraries and applications support extended terminfo binary format,
+The \fBncurses\fP libraries and applications support
+extended terminfo binary format,
 allowing users to define capabilities which are loaded at runtime.
 This
 extension is made possible by using the fact that the other implementations
@@ -280,7 +281,8 @@ A small number of terminal descriptions use uppercase characters in
 their names.
 If the underlying filesystem ignores the difference between
 uppercase and lowercase,
-\fBncurses\fP represents the \*(``first character\*('' of the terminal name used as
+\fBncurses\fP represents the \*(``first character\*(''
+of the terminal name used as
 the intermediate level of a directory tree in (two-character) hexadecimal form.
 .SH EXAMPLE
 As an example, here is a description for the Lear-Siegler
diff --git a/man/terminfo.head b/man/terminfo.head
index d3236847..24d4e2a5 100644
--- a/man/terminfo.head
+++ b/man/terminfo.head
@@ -26,7 +26,7 @@
 .\" authorization.                                                           *
 .\"***************************************************************************
 .\"
-.\" $Id: terminfo.head,v 1.32 2017/04/22 13:52:49 tom Exp $
+.\" $Id: terminfo.head,v 1.33 2018/05/19 20:57:27 tom Exp $
 .TH terminfo 5 "" "" "File Formats"
 .ds n 5
 .ds d @TERMINFO@
@@ -35,12 +35,13 @@
 .ie \n(.g .ds '' \(rq
 .el       .ds '' ''
 .de bP
-.IP \(bu 4
+.ie n  .IP \(bu 4
+.el    .IP \(bu 2
 ..
 .de NS
-.ie \n(.sp
+.ie n  .sp
 .el    .sp .5
-.ie \n(.in +4
+.ie n  .in +4
 .el    .in +2
 .nf
 .ft C			\" Courier
@@ -145,7 +146,7 @@ l l l.
 \-w	Wide mode (> 80 columns, usually 132)	vt100\-w
 .TE
 .PP
-For more on terminal naming conventions, see the \fBterm(7)\fR manual page.
+For more on terminal naming conventions, see the \fBterm\fP(7) manual page.
 .SS Terminfo Capabilities Syntax
 .PP
 The terminfo entry consists of several \fIcapabilities\fP,
diff --git a/man/terminfo.tail b/man/terminfo.tail
index 37db325e..b75da4a4 100644
--- a/man/terminfo.tail
+++ b/man/terminfo.tail
@@ -1,4 +1,4 @@
-.\" $Id: terminfo.tail,v 1.88 2017/08/12 22:26:02 tom Exp $
+.\" $Id: terminfo.tail,v 1.89 2018/05/19 21:01:52 tom Exp $
 .\" Beginning of terminfo.tail file
 .\" This file is part of ncurses.
 .\" See "terminfo.head" for copyright.
@@ -370,7 +370,7 @@ it may still be possible to craft a working
 .B nel
 out of one or both of them.
 .PP
-These capabilities suffice to describe hard-copy and \*(lqglass-tty\*(rq terminals.
+These capabilities suffice to describe hard-copy and \*(``glass-tty\*('' terminals.
 Thus the model 33 teletype is described as
 .PP
 .DT
@@ -516,12 +516,12 @@ to be sent \eE&a12c03Y padded for 6 milliseconds.
 Note that the order
 of the rows and columns is inverted here, and that the row and column
 are printed as two digits.
-Thus its \fBcup\fR capability is \*(lqcup=6\eE&%p2%2dc%p1%2dY\*(rq.
+Thus its \fBcup\fR capability is \*(``cup=6\eE&%p2%2dc%p1%2dY\*(''.
 .PP
 The Microterm \s-1ACT-IV\s0 needs the current row and column sent
 preceded by a \fB^T\fR, with the row and column simply encoded in binary,
-\*(lqcup=^T%p1%c%p2%c\*(rq.
-Terminals which use \*(lq%c\*(rq need to be able to
+\*(``cup=^T%p1%c%p2%c\*(''.
+Terminals which use \*(``%c\*('' need to be able to
 backspace the cursor (\fBcub1\fR),
 and to move the cursor up one line on the screen (\fBcuu1\fR).
 This is necessary because it is not always safe to transmit \fB\en\fR
@@ -531,7 +531,7 @@ tabs are never expanded, so \et is safe to send.
 This turns out to be essential for the Ann Arbor 4080.)
 .PP
 A final example is the \s-1LSI ADM\s0-3a, which uses row and column
-offset by a blank character, thus \*(lqcup=\eE=%p1%' '%+%c%p2%' '%+%c\*(rq.
+offset by a blank character, thus \*(``cup=\eE=%p1%' '%+%c%p2%' '%+%c\*(''.
 After sending \*(``\eE=\*('', this pushes the first parameter, pushes the
 ASCII value for a space (32), adds them (pushing the sum on the stack
 in place of the two previous values) and outputs that value as a character.
@@ -569,7 +569,7 @@ spaces to the right) these can be given as
 .BR cub ,
 .BR cuf ,
 and
-.BR cuu
+.B cuu
 with a single parameter indicating how many spaces to move.
 These are primarily useful if the terminal does not have
 .BR cup ,
@@ -700,18 +700,18 @@ either eliminated, or expanded to two untyped blanks.
 You can determine the
 kind of terminal you have by clearing the screen and then typing
 text separated by cursor motions.
-Type \*(lqabc\ \ \ \ def\*(rq using local
-cursor motions (not spaces) between the \*(lqabc\*(rq and the \*(lqdef\*(rq.
-Then position the cursor before the \*(lqabc\*(rq and put the terminal in insert
+Type \*(``abc\ \ \ \ def\*('' using local
+cursor motions (not spaces) between the \*(``abc\*('' and the \*(``def\*(''.
+Then position the cursor before the \*(``abc\*('' and put the terminal in insert
 mode.
 If typing characters causes the rest of the line to shift
 rigidly and characters to fall off the end, then your terminal does
 not distinguish between blanks and untyped positions.
-If the \*(lqabc\*(rq
-shifts over to the \*(lqdef\*(rq which then move together around the end of the
+If the \*(``abc\*(''
+shifts over to the \*(``def\*('' which then move together around the end of the
 current line and onto the next as you insert, you have the second type of
 terminal, and should give the capability \fBin\fR, which stands for
-\*(lqinsert null\*(rq.
+\*(``insert null\*(''.
 .PP
 While these are two logically separate attributes (one line versus multi-line
 insert mode, and special treatment of untyped spaces) we have seen no
@@ -959,7 +959,7 @@ give this sequence as
 If there is a way to make the cursor completely invisible, give that as
 .BR civis .
 The capability
-.BR cnorm
+.B cnorm
 should be given which undoes the effects of both of these modes.
 .PP
 If your terminal correctly generates underlined characters
@@ -1128,25 +1128,25 @@ They will be printed in the following order:
 .RS
 .TP
 run the program
-.BR iprog
+.B iprog
 .TP
 output
-.BR is1
-.BR is2
+.B is1
+.B is2
 .TP
 set the margins using
 .BR mgc ,
-.BR smgl
+.B smgl
 and
-.BR smgr
+.B smgr
 .TP
 set tabs using
 .B tbc
 and
-.BR hts
+.B hts
 .TP
 print the file
-.BR if
+.B if
 .TP
 and finally
 output
@@ -1167,7 +1167,7 @@ A set of sequences that does a harder reset from a totally unknown state
 can be given as
 .BR rs1 ,
 .BR rs2 ,
-.BR rf
+.B rf
 and
 .BR rs3 ,
 analogous to
@@ -1175,13 +1175,13 @@ analogous to
 .B is2 ,
 .B if
 and
-.BR is3
+.B is3
 respectively.
 These strings are output by the \fB@RESET@\fP program,
 which is used when the terminal gets into a wedged state.
 Commands are normally placed in
 .BR rs1 ,
-.BR rs2
+.B rs2
 .B rs3
 and
 .B rf
@@ -1196,7 +1196,7 @@ needed since the terminal is usually already in 80 column mode.
 The \fB@RESET@\fP program writes strings including
 .BR iprog ,
 etc., in the same order as the
-.IR init
+.I init
 program, using 
 .BR rs1 ,
 etc., instead of
@@ -1207,7 +1207,7 @@ If any of
 .BR rs2 ,
 .BR rs3 ,
 or
-.BR rf
+.B rf
 reset capability strings are missing, the \fB@RESET@\fP 
 program falls back upon the corresponding initialization capability string.
 .PP
diff --git a/man/tic.1m b/man/tic.1m
index cd1fd4f9..e0bf36fd 100644
--- a/man/tic.1m
+++ b/man/tic.1m
@@ -1,5 +1,5 @@
 .\"***************************************************************************
-.\" Copyright (c) 1998-2016,2017 Free Software Foundation, Inc.              *
+.\" Copyright (c) 1998-2017,2018 Free Software Foundation, Inc.              *
 .\"                                                                          *
 .\" Permission is hereby granted, free of charge, to any person obtaining a  *
 .\" copy of this software and associated documentation files (the            *
@@ -26,7 +26,7 @@
 .\" authorization.                                                           *
 .\"***************************************************************************
 .\"
-.\" $Id: tic.1m,v 1.66 2017/11/18 23:51:17 tom Exp $
+.\" $Id: tic.1m,v 1.67 2018/05/19 21:07:46 tom Exp $
 .TH @TIC@ 1M ""
 .ie \n(.g .ds `` \(lq
 .el       .ds `` ``
@@ -172,8 +172,8 @@ the string will not convert completely.
 .RE
 .TP
 \fB\-c\fR
-tells \fB@TIC@\fP to only check \fIfile\fR for errors, including syntax problems and
-bad use links.
+tells \fB@TIC@\fP to only check \fIfile\fR for errors,
+including syntax problems and bad use-links.
 If you specify \fB\-C\fR (\fB\-I\fR) with this option, the code
 will print warnings about entries which, after use resolution, are more than
 1023 (4096) bytes long.
@@ -205,7 +205,8 @@ the list, the entry will be written or translated as normal.
 Otherwise no output will be generated for it.
 The option value is interpreted as a file containing the list if it
 contains a '/'.
-(Note: depending on how tic was compiled, this option may require \fB\-I\fR or \fB\-C\fR.)
+(Note: depending on how tic was compiled,
+this option may require \fB\-I\fR or \fB\-C\fR.)
 .TP
 \fB\-f\fR
 Display complex terminfo strings which contain if/then/else/endif expressions
@@ -273,7 +274,8 @@ versions of terminfo like those on SVr1, Ultrix, or HP/UX that do not support
 the full set of SVR4/XSI Curses terminfo; and outright broken ports like AIX 3.x
 that have their own extensions incompatible with SVr4/XSI.
 Available subsets
-are "SVr1", "Ultrix", "HP", "BSD" and "AIX"; see \fBterminfo\fR(\*n) for details.
+are "SVr1", "Ultrix", "HP", "BSD" and "AIX";
+see \fBterminfo\fR(\*n) for details.
 .TP
 \fB\-r\fR
 Force entry resolution (so there are no remaining tc capabilities) even
@@ -398,7 +400,8 @@ The name field cannot
 exceed 512 bytes.
 Terminal names exceeding the maximum alias length
 (32 characters on systems with long filenames, 14 characters otherwise)
-will be truncated to the maximum alias length and a warning message will be printed.
+will be truncated to the maximum alias length
+and a warning message will be printed.
 .SH COMPATIBILITY
 There is some evidence that historic \fB@TIC@\fR implementations treated
 description fields with no whitespace in them as additional aliases or
diff --git a/man/tput.1 b/man/tput.1
index ad7e81a4..10e47b7b 100644
--- a/man/tput.1
+++ b/man/tput.1
@@ -1,6 +1,6 @@
 '\" t
 .\"***************************************************************************
-.\" Copyright (c) 1998-2016,2017 Free Software Foundation, Inc.              *
+.\" Copyright (c) 1998-2017,2018 Free Software Foundation, Inc.              *
 .\"                                                                          *
 .\" Permission is hereby granted, free of charge, to any person obtaining a  *
 .\" copy of this software and associated documentation files (the            *
@@ -27,7 +27,7 @@
 .\" authorization.                                                           *
 .\"***************************************************************************
 .\"
-.\" $Id: tput.1,v 1.57 2017/11/20 01:07:02 tom Exp $
+.\" $Id: tput.1,v 1.58 2018/05/19 21:07:46 tom Exp $
 .TH @TPUT@ 1 ""
 .ds d @TERMINFO@
 .ds n 1
@@ -244,7 +244,8 @@ Before ncurses 6.1, the two utilities were different from each other:
 (not done with \fB@TPUT@\fP).
 .bP
 On the other hand, \fB@TSET@\fP's repertoire of terminal capabilities for
-resetting the terminal was more limited, i.e., only \fBreset_1string\fP, \fBreset_2string\fP and \fBreset_file\fP
+resetting the terminal was more limited,
+i.e., only \fBreset_1string\fP, \fBreset_2string\fP and \fBreset_file\fP
 in contrast to the tab-stops and margins which are set by this utility.
 .bP
 The \fBreset\fP program is usually an alias for \fB@TSET@\fP,
@@ -328,7 +329,8 @@ variable \fBTERM\fR.
 .RE
 .TP 5
 \&
-This example shows \fB@TPUT@\fR processing several capabilities in one invocation.
+This example shows \fB@TPUT@\fR processing several capabilities
+in one invocation.
 It clears the screen,
 moves the cursor to position 10, 10
 and turns on bold (extra bright) mode.
@@ -413,7 +415,8 @@ whose \fBinit\fP and \fBreset\fP  subcommands
 (more than half the program) were incorporated from
 the \fBreset\fP feature of BSD \fBtset\fP written by Eric Allman.
 .PP
-Keith Bostic replaced the BSD \fBtput\fP command in 1989 with a new implementation
+Keith Bostic replaced the BSD \fBtput\fP command in 1989
+with a new implementation
 based on the AT&T System V program \fBtput\fP.
 Like the AT&T program, Bostic's version
 accepted some parameters named for \fIterminfo capabilities\fP
@@ -457,7 +460,8 @@ before falling back to \*(``/dev/tty\*('' and finally just assumes
 a 1200Bd terminal.
 When updating terminal modes, it ignores errors.
 .IP
-Until changes made after ncurses 6.0, \fB@TPUT@\fP did not modify terminal modes.
+Until changes made after ncurses 6.0,
+\fB@TPUT@\fP did not modify terminal modes.
 \fB@TPUT@\fP now uses a similar scheme,
 using functions shared with \fB@TSET@\fP
 (and ultimately based on the 4.4BSD \fBtset\fP).
@@ -525,11 +529,13 @@ only a few may not be apparent.
 X/Open Curses Issue 7 documents \fBtput\fP differently, with \fIcapname\fP
 and the other features used in this implementation.
 .bP
-That is, there are two standards for \fBtput\fP: POSIX (a subset) and X/Open Curses (the full implementation).
+That is, there are two standards for \fBtput\fP:
+POSIX (a subset) and X/Open Curses (the full implementation).
 POSIX documents a subset to avoid the complication of including X/Open Curses
 and the terminal capabilities database.
 .bP
-While it is certainly possible to write a \fBtput\fP program without using curses,
+While it is certainly possible to write a \fBtput\fP program
+without using curses,
 none of the systems which have a curses implementation provide
 a \fBtput\fP utility which does not provide the \fIcapname\fP feature.
 .SH SEE ALSO
diff --git a/man/tset.1 b/man/tset.1
index 14290f92..9fe56213 100644
--- a/man/tset.1
+++ b/man/tset.1
@@ -1,5 +1,5 @@
 .\"***************************************************************************
-.\" Copyright (c) 1998-2016,2017 Free Software Foundation, Inc.              *
+.\" Copyright (c) 1998-2017,2018 Free Software Foundation, Inc.              *
 .\"                                                                          *
 .\" Permission is hereby granted, free of charge, to any person obtaining a  *
 .\" copy of this software and associated documentation files (the            *
@@ -26,7 +26,7 @@
 .\" authorization.                                                           *
 .\"***************************************************************************
 .\"
-.\" $Id: tset.1,v 1.52 2017/11/20 01:27:53 tom Exp $
+.\" $Id: tset.1,v 1.53 2018/05/19 21:07:46 tom Exp $
 .TH @TSET@ 1 ""
 .ie \n(.g .ds `` \(lq
 .el       .ds `` ``
@@ -101,7 +101,8 @@ and \fBCOLUMNS\fP variables specify this),
 use this to set the operating system's notion of the window size.
 .bP
 if the \*(``\fB\-c\fP\*('' option is enabled,
-the backspace, interrupt and line kill characters (among many other things) are set
+the backspace, interrupt and line kill characters
+(among many other things) are set
 .bP
 unless the \*(``\fB\-I\fP\*('' option is enabled,
 the terminal
@@ -354,7 +355,8 @@ The \fB\-a\fR, \fB\-d\fR, and \fB\-p\fR options are similarly
 not documented or useful, but were retained as they appear to be in
 widespread use.  It is strongly recommended that any usage of these
 three options be changed to use the \fB\-m\fR option instead.
-The \fB\-a\fP, \fB\-d\fP, and \fB\-p\fR options are therefore omitted from the usage summary above.
+The \fB\-a\fP, \fB\-d\fP, and \fB\-p\fR options
+are therefore omitted from the usage summary above.
 .PP
 Very old systems, e.g., 3BSD, used a different terminal driver which
 was replaced in 4BSD in the early 1980s.
diff --git a/misc/terminfo.src b/misc/terminfo.src
index 9600dabc..0ba7d5c4 100644
--- a/misc/terminfo.src
+++ b/misc/terminfo.src
@@ -6,8 +6,8 @@
 # Report bugs and new terminal descriptions to
 #	bug-ncurses@gnu.org
 #
-#	$Revision: 1.684 $
-#	$Date: 2018/03/18 00:08:38 $
+#	$Revision: 1.699 $
+#	$Date: 2018/05/19 20:10:26 $
 #
 # The original header is preserved below for reference.  It is noted that there
 # is a "newer" version which differs in some cosmetic details (but actually
@@ -1421,10 +1421,9 @@ iTerm.app|iterm|iTerm.app terminal emulator for Mac OS X,
 	    %p9%t\016%e\017%;,
 	sgr0=\E[m\017, smacs=^N, smam=\E[?7h, smcup=\E7\E[?47h,
 	smir=\E[4h, smkx=\E[?1h\E=, smso=\E[7m, smul=\E[4m,
-	tbc=\E[3g, u6=\E[%i%d;%dR, u7=\E[6n, u8=\E[?1;2c, u9=\E[c,
-	vpa=\E[%i%p1%dd, kEND5=\E[1;5F, kHOM5=\E[1;5H,
-	use=xterm+sl-twm, use=vt100+keypad, use=xterm+x11mouse,
-	use=xterm+256setaf,
+	tbc=\E[3g, vpa=\E[%i%p1%dd, kEND5=\E[1;5F, kHOM5=\E[1;5H,
+	use=ansi+enq, use=xterm+sl-twm, use=vt100+keypad,
+	use=xterm+x11mouse, use=xterm+256setaf,
 
 # iTerm2 3.0.15
 #
@@ -1454,16 +1453,20 @@ iTerm.app|iterm|iTerm.app terminal emulator for Mac OS X,
 #
 # Italic text did not work initially, apparently because upgrading did not
 # add/change that preference (set in Preferences, Profiles, Text).  A new
-# install of iTerm 3.0.15 provides italics by default (blinking text:no).
+# install of iTerm 3.0.15 provides italics by default (blinking text is an
+# option in the preferences dialog).
 #
 # 2018/01/21: found xterm+sm+1006 did not work with version 3.1.5
+# 2018/05/19: xterm+sm+1006 seems to work with 3.1.6beta -TD
 iTerm2.app|iterm2|terminal emulator for Mac OS X,
-	blink@, dim=\E[2m, kEND=\E[1;2F, kHOM=\E[1;2H, ka1@, ka3@,
-	kb2@, kc1@, kc3@, kent@, kf13=\E[1;2P, kf14=\E[1;2Q,
-	kf15=\E[1;2R, kf16=\E[1;2S, kf17=\E[15;2~, kf18=\E[17;2~,
-	kf19=\E[18;2~, kf2=\EOQ, kf20=\E[19;2~, kf21=\E[20;2~,
-	kf22=\E[21;2~, kf23=\E[23;2~, kf24=\E[24;2~,
-	kind=\E[1;2B, kri=\E[1;2A,
+	blink=\E[5m, cbt=\E[Z, dim=\E[2m, indn=\E[%p1%dS,
+	kEND=\E[1;2F, kHOM=\E[1;2H, ka1@, ka3@, kb2@, kc1@, kc3@, kent@,
+	kf13=\E[1;2P, kf14=\E[1;2Q, kf15=\E[1;2R, kf16=\E[1;2S,
+	kf17=\E[15;2~, kf18=\E[17;2~, kf19=\E[18;2~, kf2=\EOQ,
+	kf20=\E[19;2~, kf21=\E[20;2~, kf22=\E[21;2~,
+	kf23=\E[23;2~, kf24=\E[24;2~, kind=\E[1;2B, kri=\E[1;2A,
+	nel=\EE, op=\E[39;49m, rin=\E[%p1%dT, rmso=\E[27m,
+	rmul=\E[24m, rs2=\E[!p\E[?3;4l\E[4l\E>\E[?1000l,
 	sgr=\E[0%?%p6%t;1%;%?%p2%t;4%;%?%p1%p3%|%t;7%;%?%p4%t;5%;%?
 	    %p5%t;2%;m%?%p9%t\016%e\017%;,
 	kDN3=\E\E[B, kDN4=\E[1;10B, kDN5=\E[1;5B, kDN6=\E[1;6B,
@@ -1474,7 +1477,8 @@ iTerm2.app|iterm2|terminal emulator for Mac OS X,
 	kLFT5=\E[1;5D, kLFT6=\E[1;6D, kNXT3=\E\E[6~,
 	kPRV3=\E\E[5~, kRIT3=\E\E[C, kRIT4=\E[1;10C,
 	kRIT5=\E[1;5C, kRIT6=\E[1;6C, kUP3=\E\E[A, kUP4=\E[1;10A,
-	kUP5=\E[1;5A, kUP6=\E[1;6A, use=ecma+italics, use=iterm,
+	kUP5=\E[1;5A, kUP6=\E[1;6A, use=xterm+alt+title,
+	use=ecma+italics, use=iterm,
 
 # xnuppc - Darwin PowerPC Console (a.k.a. "darwin")
 #
@@ -7002,9 +7006,8 @@ screen3|older VT 100/ANSI X3.64 virtual terminal,
 # match the terminal.
 tmux|tmux terminal multiplexer,
 	invis=\E[8m, rmso=\E[27m,
-	sgr=\E[0%?%p6%t;1%;%?%p1%t;7%;%?%p2%t;4%;%?%p3%t;7%;%?%p4%t;
-	    \s\s\s\s\s\s\s\s\s\s\s\s5%;%?%p5%t;2%;m%?%p9%t\016%e
-	    \017%;,
+	sgr=\E[0%?%p6%t;1%;%?%p2%t;4%;%?%p1%p3%|%t;7%;%?%p4%t;5%;%?
+	    %p5%t;2%;%?%p7%t;8%;m%?%p9%t\016%e\017%;,
 	smso=\E[7m, E3=\E[3J, use=ecma+italics,
 	use=ecma+strikeout, use=xterm+edit, use=xterm+pcfkeys,
 	use=xterm+sl, use=xterm+tmux, use=screen,
@@ -25580,7 +25583,7 @@ v3220|LANPAR Vision II model 3220/3221/3222,
 #	  that could be returned -TD
 #	+ added a few capabilities fixed in recent vte -TD
 #
-# 20160817
+# 2016-08-17
 #	+ correct a typo in interix -TD
 #
 # 2016-09-24
@@ -25693,23 +25696,23 @@ v3220|LANPAR Vision II model 3220/3221/3222,
 #	  match xterm #272, reflecting packager's changes -TD
 #	+ remove "boolean" Se, Ss from st-0.7 -TD
 #
-# 2017-01-04
+# 2018-01-04
 #	+ add konsole-direct and st-direct -TD
 #	+ remove unsupported "Tc" capability from st-0.7; use st-direct if
 #	  direct-colors are wanted -TD
 #
-# 2017-01-17
+# 2018-01-17
 #	+ add vte-direct -TD
 #	+ add XT, hpa, indn, and vpa to screen, and invis, E3 to tmux (patch by
 #	  Pierre Carru)
 #
-# 2017-01-21
+# 2018-01-21
 #	+ use xterm+sm+1006 in xterm-new, vte-2014 -TD
 #	+ use xterm+x11mouse in iterm, iterm2, mlterm3 because xterm's 1006
 #	  mode does not work with those programs.  konsole is debatable -TD
 #	+ add "termite" entry (report by Markus Pfeiffer) -TD
 #
-# 2017-01-27
+# 2018-01-27
 #	+ trim "XT" from screen entry -TD
 #	+ modify iterm to use xterm+sl-twm building block -TD
 #	+ mark konsole-420pc, konsole-vt100, konsole-xf3x obsolete reflecting
@@ -25735,4 +25738,10 @@ v3220|LANPAR Vision II model 3220/3221/3222,
 #	+ trim some redundant capabilities from st-0.7 -TD
 #	+ trim unnecessary setf/setb from interix -TD
 #
+# 2018-05-19
+#	+ trim spurious whitespace from tmux in 2018-02-24 changes;
+#	  fix some inconsistencies in/between tmux- and iterm2-entries for SGR
+#	  (report by C Anthony Risinger)
+#	+ improve iterm2 using some xterm features which it has adapted -TD
+#
 ######## SHANTIH!  SHANTIH!  SHANTIH!
diff --git a/ncurses/base/lib_color.c b/ncurses/base/lib_color.c
index 41496eed..409b66ca 100644
--- a/ncurses/base/lib_color.c
+++ b/ncurses/base/lib_color.c
@@ -48,7 +48,7 @@
 #define CUR SP_TERMTYPE
 #endif
 
-MODULE_ID("$Id: lib_color.c,v 1.137 2018/03/01 15:02:12 tom Exp $")
+MODULE_ID("$Id: lib_color.c,v 1.138 2018/05/16 08:24:08 tom Exp $")
 
 #ifdef USE_TERM_DRIVER
 #define CanChange      InfoOf(SP_PARM).canchange
@@ -938,9 +938,12 @@ _nc_pair_content(SCREEN *sp, int pair, int *f, int *b)
     if (!ValidPair(sp, pair)) {
 	result = ERR;
     } else {
-	int fg = FORE_OF(sp->_color_pairs[pair]);
-	int bg = BACK_OF(sp->_color_pairs[pair]);
+	int fg;
+	int bg;
 
+	_nc_reserve_pairs(sp, pair);
+	fg = FORE_OF(sp->_color_pairs[pair]);
+	bg = BACK_OF(sp->_color_pairs[pair]);
 #if NCURSES_EXT_FUNCS
 	if (isDefaultColor(fg))
 	    fg = -1;
diff --git a/package/debian-mingw/changelog b/package/debian-mingw/changelog
index c81517c7..70dc3e7c 100644
--- a/package/debian-mingw/changelog
+++ b/package/debian-mingw/changelog
@@ -1,8 +1,8 @@
-ncurses6 (6.1+20180512) unstable; urgency=low
+ncurses6 (6.1+20180519) unstable; urgency=low
 
   * latest weekly patch
 
- -- Thomas E. Dickey <dickey@invisible-island.net>  Sat, 12 May 2018 06:58:29 -0400
+ -- Thomas E. Dickey <dickey@invisible-island.net>  Tue, 15 May 2018 20:44:03 -0400
 
 ncurses6 (5.9-20131005) unstable; urgency=low
 
diff --git a/package/debian-mingw64/changelog b/package/debian-mingw64/changelog
index c81517c7..70dc3e7c 100644
--- a/package/debian-mingw64/changelog
+++ b/package/debian-mingw64/changelog
@@ -1,8 +1,8 @@
-ncurses6 (6.1+20180512) unstable; urgency=low
+ncurses6 (6.1+20180519) unstable; urgency=low
 
   * latest weekly patch
 
- -- Thomas E. Dickey <dickey@invisible-island.net>  Sat, 12 May 2018 06:58:29 -0400
+ -- Thomas E. Dickey <dickey@invisible-island.net>  Tue, 15 May 2018 20:44:03 -0400
 
 ncurses6 (5.9-20131005) unstable; urgency=low
 
diff --git a/package/debian/changelog b/package/debian/changelog
index fa12a818..57c6d410 100644
--- a/package/debian/changelog
+++ b/package/debian/changelog
@@ -1,8 +1,8 @@
-ncurses6 (6.1+20180512) unstable; urgency=low
+ncurses6 (6.1+20180519) unstable; urgency=low
 
   * latest weekly patch
 
- -- Thomas E. Dickey <dickey@invisible-island.net>  Sat, 12 May 2018 06:58:29 -0400
+ -- Thomas E. Dickey <dickey@invisible-island.net>  Tue, 15 May 2018 20:44:03 -0400
 
 ncurses6 (5.9-20120608) unstable; urgency=low
 
diff --git a/package/mingw-ncurses.nsi b/package/mingw-ncurses.nsi
index 459bc6d8..a417834c 100644
--- a/package/mingw-ncurses.nsi
+++ b/package/mingw-ncurses.nsi
@@ -1,4 +1,4 @@
-; $Id: mingw-ncurses.nsi,v 1.270 2018/05/12 10:58:29 tom Exp $
+; $Id: mingw-ncurses.nsi,v 1.271 2018/05/16 00:44:03 tom Exp $
 
 ; TODO add examples
 ; TODO bump ABI to 6
@@ -10,7 +10,7 @@
 !define VERSION_MAJOR "6"
 !define VERSION_MINOR "1"
 !define VERSION_YYYY  "2018"
-!define VERSION_MMDD  "0512"
+!define VERSION_MMDD  "0519"
 !define VERSION_PATCH ${VERSION_YYYY}${VERSION_MMDD}
 
 !define MY_ABI   "5"
diff --git a/package/mingw-ncurses.spec b/package/mingw-ncurses.spec
index 519b9913..75266735 100644
--- a/package/mingw-ncurses.spec
+++ b/package/mingw-ncurses.spec
@@ -3,7 +3,7 @@
 Summary: shared libraries for terminal handling
 Name: mingw32-ncurses6
 Version: 6.1
-Release: 20180512
+Release: 20180519
 License: X11
 Group: Development/Libraries
 Source: ncurses-%{version}-%{release}.tgz
diff --git a/package/ncurses.spec b/package/ncurses.spec
index 291eba5e..d8555f05 100644
--- a/package/ncurses.spec
+++ b/package/ncurses.spec
@@ -1,7 +1,7 @@
 Summary: shared libraries for terminal handling
 Name: ncurses6
 Version: 6.1
-Release: 20180512
+Release: 20180519
 License: X11
 Group: Development/Libraries
 Source: ncurses-%{version}-%{release}.tgz
